jobs/build/merge_ocp/Jenkinsfile

```diff
@@ -152,29 +171,30 @@ node {
                                     diffstat = sh(returnStdout: true, script: "git diff --cached --stat origin/${downstream}").trim()
 
                                     if (diffstat != "") {
-                                        echo "New commits from merge:\n${diffstat}"
+                                        echo "New commits from openshift/origin merge:\n${diffstat}"
                                         sh "git push origin ${downstream}:${downstream}"
-
-                                        if (params.SCHEDULE_INCREMENTAL && version.startsWith('4.')) {
-                                            build(
-                                                job: 'build%2Focp4',
-                                                propagate: false,
-                                                wait: false,
-                                                parameters: [
-                                                    string(name: 'BUILD_VERSION', value: version),
-                                                    booleanParam(name: 'FORCE_BUILD', value: false),
-                                                ]
-                                            )
-                                            currentBuild.description += "triggered build: ${version}\n"
-                                        }
-
+                                        scheduleBuild = true
                                     } else{
-                                        echo "${downstream} was already up to date. Nothing to do."
+                                        echo "${downstream} openshift/ose was already up to date. Nothing to merge."
                                     }
                                 }
 
                                 successful.add(version)
                                 echo "Success running ${version} merge"
+
+                                if (params.SCHEDULE_INCREMENTAL && scheduleBuild) {
+                                    build(
+                                        job: 'build%2Focp4',
```

Comment:
Actually... I don't like this. Say a change comes in. A build gets kicked off. Until that build at least gets done updating distgit, this job is going to keep kicking off more builds (which will immediately block, and presumably finish quickly doing nothing once unblocked). That just seems really untidy. There will be a lot of fluff job runs. Is there any clever way to prevent builds being scheduled when already running?

Reply:
Nevermind this, lock check above should prevent spamming ocp4.